onmt/translate/beam.py

```diff
@@ -172,12 +172,57 @@ def get_hyp(self, timestep, k):
         Walk back to construct the full hypothesis.
         """
         hyp, attn = [], []
-        for j in range(len(self.prev_ks[:timestep]) - 1, -1, -1):
+        for j in range(timestep - 1, -1, -1):
             hyp.append(self.next_ys[j + 1][k])
             attn.append(self.attn[j][k])
             k = self.prev_ks[j][k]
         return hyp[::-1], torch.stack(attn[::-1])
 
+    def update_score(self, attn):
+        if "prev_penalty" in self.global_state:
+            self.scores.add_(self.global_state["prev_penalty"])
+            penalty = self.global_scorer.cov_penalty(
+                self, self.global_state["coverage"] + attn,
```

Comment:
first self should be removed

Reply:
okay I think I know why.
now cov_penalty() takes beam.scores and cov as arguments, not the beal itself, nor the beta
Testing it and will fix this.

Reply:
well I chnaged to:
            penalty = self.global_scorer.cov_penalty(
                self.scores, self.global_state["coverage"] + attn)
But even if it is "running" apparently fine, results are not good.